pyccel/ast/builtins.py

```diff
@@ -774,6 +776,81 @@ def is_homogeneous(self):
         """
         return True
 
+#==============================================================================
+class PythonDict(PyccelFunction):
+    """
+    Class representing a call to Python's `{}` function.
+
+    Class representing a call to Python's `{}` function which generates a
+    literal Python dict. This operator does not handle `**a` expressions.
+
+    Parameters
+    ----------
+    keys : iterable[TypedAstNode]
+        The keys of the new dictionary.
+    values : iterable[TypedAstNode]
+        The values of the new dictionary.
+    """
+    __slots__ = ('_keys', '_values', '_shape', '_class_type')
+    _attribute_nodes = ('_keys', '_values')
+    _rank = 1
```

Comment:
why is the rank fixed to 1?

Reply:
What would you suggest instead? The rank of the values or the rank of the keys?

Reply:
I thought it was meant to represent the rank of the whole dictionary, for example, what would be the rank of this one:
```python
a = {
  "key1" : {
    "nestedKey1" : "nestVal1",
  }
}
```